dex/farm-staking/src/custom_rewards.rs

```diff
@@ -26,36 +26,21 @@ pub trait CustomRewardsModule:
     }
 
     fn generate_aggregated_rewards(&self) {
-        let extra_rewards = self.calculate_extra_rewards_since_last_allocation();
+        let mut extra_rewards = self.calculate_extra_rewards_since_last_allocation();
         if extra_rewards > 0 {
-            self.increase_reward_reserve(&extra_rewards);
-            self.update_reward_per_share(&extra_rewards);
-        }
-    }
-
-    fn calculate_rewards_with_apr_limit(
-        &self,
-        amount: &BigUint,
-        current_reward_per_share: &BigUint,
-        initial_reward_per_share: &BigUint,
-        last_claim_block: u64,
-    ) -> BigUint {
-        let unbounded_rewards =
-            self.calculate_reward(amount, current_reward_per_share, initial_reward_per_share);
-        if unbounded_rewards == 0u32 {
-            return unbounded_rewards;
-        }
+            let mut accumulated_rewards = self.accumulated_rewards().get();
+            let total_rewards = &accumulated_rewards + &extra_rewards;
+            let reserve = self.reward_reserve().get();
+            if total_rewards > reserve {
+                let amount_over_reserve = reserve - total_rewards;
```

Comment:
Shouldn't this be `total_rewards - reserve`? Note that `total_rewards` is the larger number.

Reply:
Right. Will fix, thanks.